File: OCH_Tao/ABC/A/123.py
L = []
for i in range(5):
  L.append(int(input()))
K = int(input())
while len(L)>1:
  x = L.pop(0)
  for i in L:
    if i-x > K:
      print(":(")
      break
  else:
    continue
  break
else:
  print("Yay!")

File: OCH_Tao/ABC/A/123.js
const input = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n")).map(x => parseInt(x, 10));
if (input[5] < input[4] - input[0]) {
  console.log(":(");
} else {
  console.log("Yay!");
}